spec/support/test_access_patch.rb

```diff
@@ -0,0 +1,27 @@
+# Give access to otherwise private members
+module Datadog
+  class Writer
+    attr_accessor :trace_handler, :service_handler, :worker
```

Comment:
I would say to only copy what we need from these helpers as we go. Honestly, I think we should never change code to be able to test it, so it might be worth revisiting some decisions made in the past as we port the tests to `rspec`.

A couple of thoughts (I might be completely wrong, just giving my personal opinion!):

* Injecting fake components is perfectly fine as long as they're not the test target (that is, we're injecting them to test the interaction with other components).
* Private methods should always be treated as implementation details; we should NEVER rely on them for writing  tests. If you can't test something you want "directly" consider testing the side effects of it;

Reply:
Totally agree with these points.

The original strategy here was just to get everything copied over from `helper.rb` so whoever is copying over old Minitests to RSpec could convert a test much more easily with backwards compatibility.

But, to your point, yeah, this probably does represent a great opportunity to rewrite tests. Maybe the compromise I'd suggest is keeping these modules in the `support` folder available, but disabled. If we convert tests that need these modules, they can be activated and refactored if appropriate at that time.

Although a module like `test_access_patch` should be considered deprecated, it will be useful to keep so any non-RSpec savvy dev can quickly convert over a test, without having the headache of converting helpers they need from `helper.rb` to `spec_helper.rb`.

Reply:
Humn, I think I agree with that. My fear is just that it might be too tempting to just copy paste stuff that make use of it if we have it available, but I would say that it is preferable to encourage migrating the tests (by making the process and easy as possible) over "re-thinking" the use of these helpers for now :)